### PR TITLE
feat(core): Add OStringStream str(), view(), put(), write(), tellp()

### PR DESCRIPTION
**Pull request by dkrutskikh**
### **PR Type**
Enhancement, Tests

___

### **Description**
- Add str() setter, view(), put(), write(), and tellp() methods to OStringStream

- Implement null character validation in FixedString push_back, append, and operator+=

- Add comprehensive test coverage for new OStringStream methods

- Configure C17 and C++23 standards across all platform targets

___

### Diagram Walkthrough

```mermaid
flowchart LR
  OStringStream["OStringStream<br/>class"]
  FixedString["FixedString<br/>class"]
  
  OStringStream -- "str setter" --> SetContent["Set content from<br/>StringLike"]
  OStringStream -- "view" --> CStringView["Return CStringView<br/>non-owning"]
  OStringStream -- "put" --> PutChar["Append single<br/>character"]
  OStringStream -- "write" --> WriteBuffer["Append multiple<br/>characters"]
  OStringStream -- "tellp" --> GetPos["Get write<br/>position"]
  
  PutChar --> FixedString
  WriteBuffer --> FixedString
  
  FixedString -- "validate" --> NullCheck["Reject null<br/>characters"]
```

<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>o_string_stream.hpp</strong><dd><code>Add stream output methods and type aliases</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/o_string_stream.hpp

<ul><li>Add type aliases <code>char_type</code> and <code>pos_type</code> for stream compatibility<br> <li> Add <code>str(const SourceStringType&)</code> setter method to replace stream <br>content<br> <li> Add <code>view()</code> method returning <code>CStringView</code> for non-owning access<br> <li> Add <code>put(char_type)</code> method to append single characters with chaining <br>support<br> <li> Add <code>write(const char_type*, size_t)</code> method to append multiple <br>characters<br> <li> Add <code>tellp()</code> method returning current write position<br> <li> Update documentation for constructor and getter methods</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/256/files#diff-5ebe7f0e4874201f5bcfb4bf28cfd06a0e007cb7d825f766cfe4b1394656c831">+113/-3</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>o_string_stream.inl</strong><dd><code>Implement new OStringStream methods</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/o_string_stream.inl

<ul><li>Implement <code>str()</code> setter using assignment operator<br> <li> Implement <code>view()</code> creating <code>CStringView</code> from internal string<br> <li> Implement <code>put()</code> delegating to <code>push_back()</code> with return reference<br> <li> Implement <code>write()</code> looping through buffer and calling <code>push_back()</code><br> <li> Implement <code>tellp()</code> returning string size</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/256/files#diff-84b520b8c1ad074985e05c62cfa95b52510fbcc5c66c0713e82eefe0a42092f1">+35/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>fixed_string.inl</strong><dd><code>Add null character validation to append methods</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/fixed_string.inl

<ul><li>Add null character validation in <code>push_back()</code> with early return<br> <li> Add null character validation in <code>append(char, size_t)</code> with early <br>return<br> <li> Add null character validation in <code>operator+=(char)</code> with early return<br> <li> Prevent null characters from being appended to strings</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/256/files#diff-d04929dc1cdf245bb0c31aceff54621fa42e4dc2743fd41a5021ebe40fb7dfe2">+13/-1</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>o_string_stream.test.cpp</strong><dd><code>Add comprehensive tests for new OStringStream methods</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/core/o_string_stream.test.cpp

<ul><li>Add test suite for <code>str()</code> setter with FixedString, CStringView, and <br>std::string<br> <li> Add test suite for <code>view()</code> method covering content, empty streams, and <br>updates<br> <li> Add test suite for <code>put()</code> method with chaining and special characters<br> <li> Add test suite for <code>write()</code> method with partial strings and binary data<br> <li> Add test suite for <code>tellp()</code> method with compile-time and runtime checks</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/256/files#diff-c9b47407e3150512eae77a5c23fcef5413e04cba702876df0a901f0686ae241f">+305/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>fixed_string.test.cpp</strong><dd><code>Update FixedString tests for null validation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/core/fixed_string.test.cpp

<ul><li>Remove test case attempting to push null character<br> <li> Update expected size from 8 to 7 after removing null character push</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/256/files#diff-7a7dda49df0493402cc2812260d4aca9f248231ac3cf96780cdb1b3c819cad4d">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ConfigureCompiler.cmake</strong><dd><code>Configure C17 and C++23 for all platforms</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

cmake/ConfigureCompiler.cmake

<ul><li>Add C17 and C++23 standard flags for Linux Desktop platform<br> <li> Add C17 and C++23 standard flags for Game Boy Advance platform<br> <li> Add C17 and C++23 standard flags for Nintendo DS platform<br> <li> Add C17 and C++23 standard flags for Nintendo 3DS and Switch platforms</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/256/files#diff-e65655878b2e7518b76a2b0a036285f2df3e0e0d5bcd32ff58b0b0790731fcc9">+17/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>ConfigureEngine.cmake</strong><dd><code>Enable C17 standard globally</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

cmake/ConfigureEngine.cmake

<ul><li>Add global C17 standard configuration with required flag<br> <li> Add C extensions disabled for C standard</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/256/files#diff-846d10958e3c1d8353e9f0e674c4ab8c866176645223f7217047473c68479039">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by dkrutskikh** [comment]
@coderabbit please suggest PR title

File: include/core/o_string_stream.inl [hunk 1]
```diff
@@ -42,6 +42,41 @@ constexpr const StringType & OStringStream<StringType>::str() const noexcept {
...
+constexpr CStringView OStringStream<StringType>::view() const noexcept {
+  return CStringView(_string.c_str());
+}
+
+template <typename StringType>
+inline OStringStream<StringType> & OStringStream<StringType>::put(char_type character) noexcept {
+  _string.append(character);
+
+  return *this;
+}
```
[the remark this answers is not in the document]
Comment (dkrutskikh) [reply include/core/o_string_stream.inl]: fixed
